component/src/main/java/org/wso2/extension/siddhi/io/websocket/source/websocketserver/WebSocketServerSource.java

```diff
@@ -0,0 +1,208 @@
+                     ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+        this.host = optionHolder.validateAndGetStaticValue(WebSocketProperties.HOST);
+        this.port = Integer.parseInt(optionHolder.validateAndGetStaticValue(WebSocketProperties.PORT));
+        String subProtocolString = optionHolder.validateAndGetStaticValue(WebSocketProperties.SUB_PROTOCOL,
+                                                                          null);
+        if (subProtocolString != null) {
+            subProtocols = WebSocketUtil.getSubProtocol(subProtocolString);
+        }
+        this.idleTimeout = Integer.parseInt(optionHolder.validateAndGetStaticValue(WebSocketProperties.IDLE_TIMEOUT,
+                                                                                   "-1"));
+        this.isTlsEnabled = Boolean.parseBoolean(optionHolder.validateAndGetStaticValue
+                (WebSocketProperties.TLS_ENABLED, "false"));
+        this.tlsKeystorePath = optionHolder.validateAndGetStaticValue(WebSocketProperties.TLS_KEYSTORE_PATH,
+                                                                      configReader.readConfig
+                                                                              (WebSocketProperties.TLS_KEYSTORE_PATH,
+                                                                               WebSocketProperties
+                                                                                       .DEFAULT_KEYSTORE_FILE_PATH));
+        this.tlsKeystorePassword = optionHolder.validateAndGetStaticValue(WebSocketProperties.TLS_KEYSTORE_PASS,
+                                                                          configReader.readConfig
+                                                                                  (WebSocketProperties
+                                                                                           .TLS_KEYSTORE_PASS,
+                                                                                   WebSocketProperties
+                                                                                           .DEFAULT_KEYSTORE_PASS));
+        this.sourceEventListener = sourceEventListener;
+    }
+
+    @Override
+    public Class[] getOutputEventClasses() {
+        return new Class[]{String.class, ByteBuffer.class};
+    }
+
+    @Override
+    public void connect(ConnectionCallback connectionCallback) throws ConnectionUnavailableException {
+        try {
+            websocketServer = new WebSocketServer(host, port, subProtocols, idleTimeout, isTlsEnabled,
+                                                  tlsKeystorePath, tlsKeystorePassword, sourceEventListener);
+            websocketServer.start();
+        } catch (InterruptedException e) {
+            throw new ConnectionUnavailableException("Error while starting the WebSocket server defined in "
+                                                             + sourceEventListener + ".", e);
```

Comment:
I think we can get the stream definition from the sourceEventListener object to include in the message

Reply:
fixed in 39e7381